src/tests/ftest/dfuse/pil4dfs_dcache.py

```diff
@@ -0,0 +1,360 @@
+                "dcache_miss": 0,
+                "no_dcache_new": 2,
+                "no_dcache_del": 2,
+                "op_name": None
+            },
+            {
+                "test_name": "test_dup",
+                "test_id": 5,
+                "dcache_add": 0,
+                "dcache_del": 0,
+                "dcache_hit": 0,
+                "dcache_miss": 0,
+                "no_dcache_new": 2,
+                "no_dcache_del": 2,
+                "op_name": None
+            }
+        ],
+    }
+
+    __dcache_re__ = {
+        "dcache_add": re.compile(r'^.+ il +DBUG .+ dcache_add\(\) .+$'),
+        "dcache_del": re.compile(r'^.+ il +DBUG .+ dcache_rec_free\(\) .+$'),
+        "dcache_hit": re.compile(r'^.+ il +DBUG .+ dcache hit:.+$'),
+        "dcache_miss": re.compile(r'^.+ il +DBUG .+ dcache miss:.+$'),
+        "no_dcache_new": re.compile(r'^.+ il +DBUG .+ dcache_find_insert_dact\(\).+$'),
+        "no_dcache_del": re.compile(r'^.+ il +DBUG .+ drec_del_at_dact\(\) .+$')
+    }
+
+    __start_test_re__ = re.compile(r'^-- START of test_.+ --$')
+    __end_test_re__ = re.compile(r'^-- END of test_.+ --$')
+
+    def setUp(self):
+        """Set up each test case."""
+        # obtain separate logs
+        self.update_log_file_names()
+
+        # Start the servers and agents
+        super().setUp()
+
+        self.log.info("Creating DAOS pool")
+        self.add_pool()
+        self.log.debug("Created DAOS pool %s", str(self.pool))
+
+        self.log.info("Creating DAOS container")
+        self.add_container(self.pool)
+        self.log.debug("Created DAOS container %s", str(self.container))
+
+        self.log.info("Mounting DFuse mount point")
+        self.start_dfuse(self.hostlist_clients, self.pool, self.container)
+        self.log.debug("Mounted DFuse mount point %s", self.dfuse.mount_dir.value)
+
+    def check_result(self, test_case, lines):
+        """Check consistency of the test output.
+
+        Args:
+            test_case (dict): dictionary of the expected output result
+            lines (list): list of the test command stdout lines
+        """
+        dcache_count = {key: 0 for key in Pil4dfsDcache.__dcache_re__}
+        op_re = None
+        if test_case['op_name'] is not None:
+            op_re = re.compile(r'^\[' + test_case['op_name'] + r'\s*\]\s+(\d+)$')
+        state = 'init'
+        for line in lines:
+            if state == 'init':
+                if Pil4dfsDcache.__start_test_re__.match(line):
+                    state = 'check_dcache'
+
+            elif state == 'check_dcache':
+                for dcache_name, dcache_re in Pil4dfsDcache.__dcache_re__.items():
+                    if dcache_re.match(line):
+                        dcache_count[dcache_name] += 1
+                if Pil4dfsDcache.__end_test_re__.match(line):
+                    for key in Pil4dfsDcache.__dcache_re__:
+                        self.assertEqual(test_case[key], dcache_count[key])
```

Comment:
Please add a descriptive message (third argument) for asserts so failures are clear

Reply:
- [ ] Add assertion error message

Reply:
Fixed with commit eace57b3b479abd420dec07bf85cc46b4a1c9c31